xbmc/windowing/gbm/DRMUtils.cpp

```diff
@@ -128,6 +128,77 @@ drm_fb * CDRMUtils::DrmFbGetFromBo(struct gbm_bo *bo)
   return fb;
 }
 
+static bool GetProperties(int fd, uint32_t id, uint32_t type, struct drm_object *object)
+{
+  drmModeObjectPropertiesPtr props = drmModeObjectGetProperties(fd, id, type);
+  if (!props)
+    return false;
+
+  object->id = id;
+  object->type = type;
+  object->props = props;
+  object->props_info = new drmModePropertyPtr[props->count_props];
+
+  for (uint32_t i = 0; i < props->count_props; i++)
+    object->props_info[i] = drmModeGetProperty(fd, props->props[i]);
+
+  return true;
+}
+
+static void FreeProperties(struct drm_object *object)
+{
+  if (object->props_info)
+  {
+    for (uint32_t i = 0; i < object->props->count_props; i++)
+      drmModeFreeProperty(object->props_info[i]);
+
+    delete [] object->props_info;
+    object->props_info = nullptr;
+  }
+
+  drmModeFreeObjectProperties(object->props);
+  object->props = nullptr;
+  object->type = 0;
+  object->id = 0;
+}
+
+static uint32_t GetPropertyId(struct drm_object *object, const char *name)
+{
+  for (uint32_t i = 0; i < object->props->count_props; i++)
+    if (!strcmp(object->props_info[i]->name, name))
+      return object->props_info[i]->prop_id;
+
+  return 0;
+}
+
+bool CDRMUtils::AddProperty(drmModeAtomicReqPtr req, struct drm_object *object, const char *name, uint64_t value)
+{
+  uint32_t property_id = GetPropertyId(object, name);
+  if (property_id)
+  {
+    int ret = drmModeAtomicAddProperty(req, object->id, property_id, value);
+    if (ret > 0)
```

Comment:
can you switch this to `ret < 0` then log and return false. Many of the other functions in here return false early on error, not return true early on success :)